proto/oracle/v1/oracle.proto

```diff
@@ -130,6 +134,16 @@ message ExchangeRateTuple {
   ];
 }
 
+message DatedPrice {
+  string exchange_rate = 1 [
+    (gogoproto.moretags) = "yaml:\"exchange_rate\"",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+
+  uint64 block_set = 2 [ (gogoproto.moretags) = "yaml:\"block_set\"" ];
```

Comment:
Maybe rename to `created_block`? `block_set` has the implication of a set data structure.